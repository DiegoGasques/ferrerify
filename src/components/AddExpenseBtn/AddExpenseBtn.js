import React from "react";
import { Link } from "react-router-dom";
import "./AddExpenseBtn.scss";

export const AddExpenseBtn = () => {
  return (
    <Link to="/new">
      <button className="AddExpenseBtn">Add New Expense!</button>
    </Link>
  );
};

export default AddExpenseBtn;
