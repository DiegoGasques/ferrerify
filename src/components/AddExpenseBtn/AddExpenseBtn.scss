.AddExpenseBtn {
  width: 100%;
  padding: $m-size;
  text-transform: uppercase;
  background: blueviolet;
  color: $snow;
  font-size: $h4;
  border: none;
  font-weight: bold;
  letter-spacing: 0.2rem;
}

.AddExpenseBtn:hover {
  box-shadow: $btn-shadow-hover;
}

.AddExpenseBtn:active {
  box-shadow: $btn-shadow-active;
}

.AddExpenseBtn:active,
.AddExpenseBtn:focus {
  outline: none;
}
