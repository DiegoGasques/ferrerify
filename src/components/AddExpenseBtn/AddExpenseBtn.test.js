import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { AddExpenseBtn } from "./AddExpenseBtn";

describe("[COMPONENT] AddExpenseBtn", () => {
  it("Tests for changes in the rendered component", () => {
    const wrapper = shallow(<AddExpenseBtn />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
