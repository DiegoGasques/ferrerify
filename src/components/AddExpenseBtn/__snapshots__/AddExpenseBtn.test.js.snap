// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] AddExpenseBtn Tests for changes in the rendered component 1`] = `
<Link
  data-testid="new"
  to="/new"
>
  <button
    className="AddExpenseBtn"
  >
    Add New Expense!
  </button>
</Link>
`;
